src/DynamoCore/Extensions/ExtensionLoader.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Dynamo.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml;
+
+namespace Dynamo.Extensions
+{
+    public class ExtensionLoader: IExtensionLoader
+    {
+        private IExtension Load(ExtensionDefinition extension)
+        {
+            try
+            {
+                var assembly = Assembly.Load(extension.AssemblyName);
+                var result = assembly.CreateInstance(extension.TypeName) as IExtension;
+                return result;
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        public IEnumerable<IExtension> Load(string extensionsPath)
+        {
```

Comment:
Let's have this return a single `IExtension` instead of scanning the whole path.  Then, let's add an overload, `LoadDirectory` that does this.  That way, we have the ability to load a single `Extension` at a time.